### PR TITLE
Add custom deadline shift for agencies

### PR DESCRIPTION
**Pull request by konstyefremov**

### CODE REVIEW

File: HappyTravel.Edo.Data/Agents/AgencyAccommodationBookingSettings.cs [hunk 1]
```diff
@@ -16,5 +16,6 @@ public class AgencyAccommodationBookingSettings
         public bool IsSupplierVisible { get; set; }
         
         public bool AreTagsVisible { get; set; }
+        public int? CustomDeadlineShift { get; set; }
```
Comment (mingazhev) [inline HappyTravel.Edo.Data/Agents/AgencyAccommodationBookingSettings.cs:19]: Please add validation for this field, it must be less than zero

Comment (konstyefremov) [reply HappyTravel.Edo.Data/Agents/AgencyAccommodationBookingSettings.cs:20]: Done

File: Api/AdministratorServices/AgencySystemSettingsManagementService.cs [hunk 2]
```diff
@@ -58,6 +58,9 @@ public async Task<Result> SetAvailabilitySearchSettings(int agencyId, AgencyAcco
 
             async Task<Result> SetSettings()
             {
+                if (settings.CustomDeadlineShift.HasValue && settings.CustomDeadlineShift >= 0) 
+                    return Result.Failure("CustomDeadlineShift should be a negative integer");
```
Comment (mingazhev) [inline Api/AdministratorServices/AgencySystemSettingsManagementService.cs:62]: must be

Comment (mingazhev) [reply Api/AdministratorServices/AgencySystemSettingsManagementService.cs:62]: And this message is too technical, need more human-readable message.
Something like `Deadline shift cannot be less than zero` or something similar

**Review by mingazhev — APPROVED** [review]
With small comment

File: Api/AdministratorServices/AgencySystemSettingsManagementService.cs (hunk 2 above)
Comment (konstyefremov) [reply Api/AdministratorServices/AgencySystemSettingsManagementService.cs]: Added a small comment and changed failure message to a more readable.

**Pull request merged**